src/main/java/io/airlift/slice/Slice.java

```diff
@@ -311,6 +326,15 @@ public long getRetainedSize()
     }
 
     /**
+     * A slice is considered compact if the base object is an array and it contains the whole array.
+     * As a result, it cannot be a view of a bigger slice.
+     */
+    public boolean isCompact()
+    {
+        return compact;
```

Comment:
Not sure that would work. If the Slice was created from a ByteBuffer, `reference` needs to point to the ByteBuffer.